modules/nw-metalLB-upgrading-operator.adoc

```diff
@@ -0,0 +1,83 @@
+// Module included in the following assemblies:
+//
+// * networking/metallb/metallb-operator-install.adoc
+
+:_content-type: PROCEDURE
+
+[id="upgrading-metalLB-configured-for-a-specific-namespace_{context}"]
+= Upgrading MetalLB when previously installed to a specific namespace
+
+Follow this procedure to allow upgrade if the Operator has previously been installed to a specific namespace.
+
+.Prerequisites
+
+* Install the {product-title} CLI (oc).
+* Log in to the OpenShift cluster as a user with `cluster-admin` privileges.
+
+.Procedure
+
+. Edit the MetalLB `OperatorGroup` CR and remove the `spec` element that contains the `targetNamespaces` entry by running the following command:
++
+[source,terminal]
+----
+$ oc patch operatorgroup -n metallb-system metallb-system --type json -p '[{ "op": "remove", "path": "/spec" }]'
```

Comment:
cc @evgenLevin @asood-rh

Reply:
The uninstall of 4.10 operator involves removing of csv and  previous subscription, that causes the operator group to be removed so we need to create operator group again. Subscription in metallb-system from 4.10 also points to stable (on 4.10 ocp cluster stable is 4.0 metallb operator).

Reply:
> The uninstall of 4.10 operator involves removing of csv and previous subscription, that causes the operator group to be removed so we need to create operator group again. Subscription in metallb-system from 4.10 also points to stable (on 4.10 ocp cluster stable is 4.0 metallb operator).

Are you sure about the stable channel on 4.10? We are publishing to the 4.10 here https://github.com/openshift/metallb-operator/blob/release-4.10/manifests/metallb-operator.package.yaml#L3 and that's what we have in the docs too: https://docs.openshift.com/container-platform/4.10/networking/metallb/metallb-operator-install.html

Reply:
@fedepaol This is what I have seen since 4.9, once you have newer version of operator, it is obtained by subscribing to channel 'stable', previously stable channel gets a version. E.g. when it was only 4.9, it could be obtained by subscribing to stable channel. Once 4.10 was packaged, there are two channels - 4.9 and stable and now it will be 4.9, 4.10 and stable.

Reply:
But the doc we are giving to users say to use 4.10 for 4.10, not stable. Plus, we didn't have stable in the operator config in 4.9 / 4.10
See https://github.com/openshift/metallb-operator/tree/release-4.10/manifests and https://github.com/openshift/metallb-operator/tree/release-4.9/manifests so I am not sure where that "stable" was coming from.

Reply:
Found it, see on slack. In 4.10 we have the option to choose, in 4.11 we'll only have "stable"